test/unittesting/utils.go

```diff
@@ -0,0 +1,652 @@
+	var result = []string{}
+	for i := 0; i < len(a1); i++ {
+		there := false
+		for y := 0; y < len(a2); y++ {
+			if a1[i] == a2[y] {
+				there = true
+			}
+		}
+		if !there {
+			result = append(result, a1[i])
+		}
+	}
+	return result
+}
+
+func getCurrTimeInMS() uint64 {
+	return uint64(time.Now().UnixNano() / 1000000)
+}
+
+func Shellout(command string) (string, string, error) {
+	installationPath := getInstallationDir()
+	var stdout bytes.Buffer
+	var stderr bytes.Buffer
+	cmd := exec.Command("bash", "-c", command)
+	cmd.Stdout = &stdout
+	cmd.Stderr = &stderr
+	cmd.Dir = installationPath
+	err := cmd.Start()
+	return stdout.String(), stderr.String(), err
+}
+
+func stopST(pid string) {
+	installationPath := getInstallationDir()
+	pidsBefore := getListOfPids()
+	if len(pidsBefore) == 0 {
+		return
+	}
+	cmd := exec.Command("kill", pid)
+	cmd.Dir = installationPath
+	err := cmd.Run()
+	if err != nil {
+		log.Fatal(err.Error())
+	}
+	startTime := getCurrTimeInMS()
+	for getCurrTimeInMS()-startTime < 10000 {
+		pidsAfter := getListOfPids()
+		if itemExists(pidsAfter, pid) {
+			time.Sleep(100 * time.Millisecond)
+		} else {
+			return
+		}
+	}
+	panic("Could not stop ST")
+}
+
+func itemExists(arr []string, item string) bool {
+	for i := 0; i < len(arr); i++ {
+		if arr[i] == item {
+			return true
+		}
+	}
+	return false
+}
+
+func cleanST() {
+	installationPath := getInstallationDir()
+	cmd := exec.Command("rm", "config.yaml")
+	cmd.Dir = installationPath
+	err := cmd.Run()
+	if err != nil {
+		log.Fatalf(err.Error(), "could not delete the config yaml file")
+	}
+
+	cmd = exec.Command("rm", "-rf", ".webserver-temp-*")
+	cmd.Dir = installationPath
+	err = cmd.Run()
+	if err != nil {
+		log.Fatalf(err.Error(), "could not delete the webserver-temp files")
+	}
+
+	cmd = exec.Command("rm", "-rf", ".started")
+	cmd.Dir = installationPath
+	err = cmd.Run()
+	if err != nil {
+		log.Fatalf(err.Error(), "could not delete the .started file")
+	}
+}
+
+func resetAll() {
+	supertokens.ResetForTest()
+	emailpassword.ResetForTest()
+	session.ResetForTest()
+	thirdparty.ResetForTest()
+	thirdpartyemailpassword.ResetForTest()
+	openid.ResetForTest()
+	passwordless.ResetForTest()
+	jwt.ResetForTest()
+}
+
+func killAllST() {
+	pids := getListOfPids()
+	for i := 0; i < len(pids); i++ {
+		stopST(pids[i])
+	}
+	resetAll()
+}
+
+func BeforeEach() {
+	killAllST()
+	setUpST()
+}
+
+func AfterEach() {
+	resetAll()
+	killAllST()
+	cleanST()
+}
+
+func returnNumberOfDirsToGoUpFromCurrentWorkingDir() string {
+	mydir, err := os.Getwd()
+	if err != nil {
+		log.Fatal(err.Error())
+	}
+	arr := strings.Split(mydir, "/")
+	counter := 0
+	for i := 0; i < len(arr); i++ {
+		if arr[i] == "supertokens-golang" {
+			counter = i
+			break
+		}
+	}
+	numberOfElems := len(arr) - counter
+	var dirUpSlash string
+	for i := 0; i < numberOfElems; i++ {
+		dirUpSlash += "../"
+	}
+	return dirUpSlash
+}
+
+func returnNumberOfSlashesRequiredToGoToRootOfTheProject() string {
```

Comment:
This seems to be redundant cause we already have `returnNumberOfDirsToGoUpFromCurrentWorkingDir`. Why is this here?